### PR TITLE
Feat/stargazers

### LINKED ISSUE #130: Nightly cron job to update star counts
This should be done by querying the GitHub API with an API token.

### PR DESCRIPTION
**Pull request by wperron**
Still working on integration tests

Fixes #130

### CODE REVIEW

**Comment by wperron** [comment]
@lucacasonato gonna need your help to test this; I don't have enough data in my staging env 😛

**Comment by wperron** [comment]
One thing to note: the Event passed to the handler isn't used by the code (there's really nothing interesting in there anyway, it's just a scheduled event), so the Lambda can be invoked directly by passing it `{}` as the event object (either in a script or in the console via the "test" button), so that's useful to re-trigger the function for debugging purposes for example.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -59,6 +59,8 @@ jobs:
             -var backup_region=$AWS_BACKUP_REGION \
             -var cloudflare_account_id=$CLOUDFLARE_ACCOUNT_ID \
             -var cloudflare_zone_id=$CLOUDFLARE_ZONE_ID \
+            -var github_username=lucacasonato \
+            -var github_toke=${{ secrets.PERSONAL_TOKEN }} \
```
Comment (lucacasonato) [inline .github/workflows/ci.yml:63]:
```suggestion
            -var github_token=${{ secrets.PERSONAL_TOKEN }} \
```

File: api/async/stargazers_test.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    await database.saveModule(ltest);
+    await database.saveModule(utest);
+
+    await handler(
+      createScheduledEvent(),
+      createContext(),
+    );
+
+    const updated = await database.getModule(ltest.name);
+    assertEquals(updated?.star_count, 2);
```
Comment (lucacasonato) [inline api/async/stargazers_test.ts:47]: It is too easy to break this test. Maybe `assert(updated?.star_count >= 2)`. That should hold true as long as me and you have starred it :-D

File: terraform/stargazers.tf [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  filename      = data.archive_file.stargazers_zip.output_path
+  function_name = "${local.prefix}_scrape_stargazers_${local.short_uuid}"
+  role          = aws_iam_role.lambda_exec_role.arn
+  handler       = "bundle.handler"
+  publish       = true
+
+  source_code_hash = filebase64sha256(data.archive_file.stargazers_zip.output_path)
+
+  runtime = "provided"
+  layers  = [aws_lambda_layer_version.deno_layer.arn, "arn:aws:lambda:${var.region}:553035198032:layer:git-lambda2:6"]
```
Comment (lucacasonato) [inline terraform/stargazers.tf:25]:
```suggestion
  layers  = [aws_lambda_layer_version.deno_layer.arn]
```

File: terraform/stargazers.tf [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  timeout     = 300
+  memory_size = 1024
+
+  environment {
+    variables = {
+      "DENO_UNSTABLE"    = "1"
+      "HANDLER_EXT"      = "js"
+      "MONGO_URI"        = var.mongodb_uri
+      "STORAGE_BUCKET"   = aws_s3_bucket.storage_bucket.id
+      "REMOTE_URL"       = "https://deno.land/x/%m@%v"
```
Comment (lucacasonato) [inline terraform/stargazers.tf:36]: Not needed for this lambda

**Review by lucacasonato — COMMENTED** [review]
Very nice

File: api/async/stargazers_test.ts (hunk 2 above)
Comment (wperron) [reply api/async/stargazers_test.ts:47]: yes I know 🤣 do you know of way to lock the number of starts for a repo? I guess worst case scenario we can simply check that the number of stars is greater than 0

**Comment by wperron** [comment]
having some repeatability issues with the localstack image (used to mock SSM) atm

File: api/async/stargazers_test.ts (hunk 2 above)
Comment (lucacasonato) [reply api/async/stargazers_test.ts:47]: I would just change it to `assert(updated?.star_count >= 2)` or `assert(updated?.star_count >= 1)`

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -59,6 +59,8 @@ jobs:
             -var backup_region=$AWS_BACKUP_REGION \
             -var cloudflare_account_id=$CLOUDFLARE_ACCOUNT_ID \
             -var cloudflare_zone_id=$CLOUDFLARE_ZONE_ID \
+            -var github_username=lucacasonato \
+            -var github_token=${{ secrets.PERSONAL_TOKEN }} \
```
Comment (lucacasonato) [inline .github/workflows/ci.yml:63]:
```suggestion
            -var github_token=${{ secrets.GH_TOKEN }} \
```

**Comment by lucacasonato** [comment]
I tested this, and with a few changes it is working. I mainly had to switch how we get the modules from the database, because we actually need the entire database entry, not just the summary we get from listModules. Otherwise things like created_at, and is_unlisted would be overwritten.

**Comment by lucacasonato** [comment]
Thanks @wperron!

**Pull request merged**